lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -278,6 +278,20 @@ def acquired(self) -> bool:  # noqa: C901
                     else:
                         logger.exception("Error creating OpenSearch lock document")
                         return False
+                else:
+                    # Ensure write was successful on all nodes
+                    # "It is important to note that this setting [`wait_for_active_shards`] greatly
+                    # reduces the chances of the write operation not writing to the requisite
+                    # number of shard copies, but it does not completely eliminate the possibility,
+                    # because this check occurs before the write operation commences. Once the
+                    # write operation is underway, it is still possible for replication to fail on
+                    # any number of shard copies but still succeed on the primary. The `_shards`
+                    # section of the write operation’s response reveals the number of shard copies
+                    # on which replication succeeded/failed."
+                    # from https://www.elastic.co/guide/en/elasticsearch/reference/8.13/docs-index_.html#index-wait-for-active-shards
+                    if response["_shards"]["failed"] > 0:
+                        logger.error("Failed to write OpenSearch lock document to all nodes")
+                        return False
```

Comment:
(Addressed on MM - allocation related questions should be solved in: https://github.com/canonical/opensearch-operator/pull/175)

Reply:
We will rely on both, allocation is broader than the index lock, it also concerns users' indices.

Reply:
Because the PR implementing the blocking allocation exclusion is not merged yet :D so until it('s merged we need to have `succeeded < len(get_nodes())` and release / delete the said doc if that's not True